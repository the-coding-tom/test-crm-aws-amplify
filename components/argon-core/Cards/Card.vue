<template>
  <div
    :class="[
      {'card-lift--hover': hover},
      {'shadow': shadow},
      {[`shadow-${shadowSize}`]: shadowSize},
      {[`bg-gradient-${gradient}`]: gradient},
      {[`bg-${type}`]: type}
    ]"
    class="card">

    <slot name="image"/>
    <div
      v-if="$slots.header"
      :class="headerClasses"
      class="card-header">
      <slot name="header"/>
    </div>
    <div
      v-if="!noBody"
      :class="bodyClasses"
      class="card-body">
      <slot/>
    </div>

    <slot v-if="noBody"/>

    <div
      v-if="$slots.footer"
      :class="footerClasses"
      class="card-footer">
      <slot name="footer"/>
    </div>
  </div>
</template>
<script>
export default {
  name: 'Card',
  props: {
    type: {
      type: String,
      description: 'Card type',
      default: null
    },
    gradient: {
      type: String,
      description: 'Card background gradient type (warning,danger etc)',
      default: null
    },
    hover: {
      type: Boolean,
      description: 'Whether card should move on hover',
      default: null
    },
    shadow: {
      type: Boolean,
      description: 'Whether card has shadow',
      default: null
    },
    shadowSize: {
      type: String,
      description: 'Card shadow size',
      default: null
    },
    noBody: {
      type: Boolean,
      default: false,
      description: 'Whether card should have wrapper body class'
    },
    bodyClasses: {
      type: [String, Object, Array],
      description: 'Card body css classes',
      default: null
    },
    headerClasses: {
      type: [String, Object, Array],
      description: 'Card header css classes',
      default: null
    },
    footerClasses: {
      type: [String, Object, Array],
      description: 'Card footer css classes',
      default: null
    }
  }
}
</script>
<style>
</style>
