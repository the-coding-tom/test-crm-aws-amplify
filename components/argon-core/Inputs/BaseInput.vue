<template>
  <div class="form-group">
    <slot name="label">
      <label
        v-if="label"
        :class="labelClasses">
        {{ label }}
      </label>
    </slot>
    <div
      :class="[
        {'input-group': hasIcon},
        {'focused': focused},
        {'input-group-alternative': alternative},
        {'has-label': label || $slots.label},
        inputGroupClasses
    ]">
      <div
        v-if="prependIcon || $slots.prepend"
        class="input-group-prepend">
        <span class="input-group-text">
          <slot name="prepend">
            <i :class="prependIcon"/>
          </slot>
        </span>
      </div>
      <slot v-bind="slotData">
        <input
          :value="value"
          :type="type"
          v-bind="$attrs"
          :valid="!error"
          :required="required"
          :class="[{'is-valid': valid === true}, {'is-invalid': error}, inputClasses]"
          class="form-control"
          v-on="listeners">
      </slot>
      <div
        v-if="appendIcon || $slots.append"
        class="input-group-append">
        <span class="input-group-text">
          <slot name="append">
            <i :class="appendIcon"/>
          </slot>
        </span>
      </div>
      <slot name="infoBlock"/>
      <slot name="error">
        <div
          v-if="error"
          class="invalid-feedback"
          style="display: block;">
          {{ error }}
        </div>
      </slot>
      <slot name="success">
        <div
          v-if="!error && valid"
          class="valid-feedback">
          {{ successMessage }}
        </div>
      </slot>
    </div>
  </div>
</template>
<script>
export default {
  name: 'BaseInput',
  inheritAttrs: false,
  props: {
    required: {
      type: Boolean,
      description: 'Whether input is required (adds an asterix *)',
      default: true
    },
    group: {
      type: Boolean,
      description:
        'Whether input is an input group (manual override in special cases)',
      default: null
    },
    valid: {
      type: Boolean,
      description: 'Whether is valid',
      default: undefined
    },
    alternative: {
      type: Boolean,
      description: 'Whether input is of alternative layout',
      default: false
    },
    label: {
      type: String,
      description: 'Input label (text before input)',
      default: null
    },
    error: {
      type: String,
      description: 'Input error (below input)',
      default: null
    },
    successMessage: {
      type: String,
      description: 'Input success message',
      default: 'Looks good!'
    },
    labelClasses: {
      type: String,
      description: 'Input label css classes',
      default: 'form-control-label'
    },
    inputClasses: {
      type: String,
      description: 'Input css classes',
      default: null
    },
    inputGroupClasses: {
      type: String,
      description: 'Input group css classes',
      default: null
    },
    value: {
      type: [String, Number],
      description: 'Input value',
      default: null
    },
    type: {
      type: String,
      description: 'Input type',
      default: 'text'
    },
    appendIcon: {
      type: String,
      description: 'Append icon (right)',
      default: null
    },
    prependIcon: {
      type: String,
      description: 'Prepend icon (left)',
      default: null
    }
  },
  data() {
    return {
      focused: false
    }
  },
  computed: {
    listeners() {
      return {
        ...this.$listeners,
        input: this.updateValue,
        focus: this.onFocus,
        blur: this.onBlur
      }
    },
    slotData() {
      return {
        focused: this.focused,
        error: this.error,
        ...this.listeners
      }
    },
    hasIcon() {
      const { append, prepend } = this.$slots
      return (
        append !== undefined ||
        prepend !== undefined ||
        this.appendIcon !== undefined ||
        this.prependIcon !== undefined ||
        this.group
      )
    }
  },
  methods: {
    updateValue(evt) {
      let value = evt.target.value
      this.$emit('input', value)
    },
    onFocus(evt) {
      this.focused = true
      this.$emit('focus', evt)
    },
    onBlur(evt) {
      this.focused = false
      this.$emit('blur', evt)
    }
  }
}
</script>
<style>
</style>
