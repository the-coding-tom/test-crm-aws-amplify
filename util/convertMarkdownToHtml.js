export const convertMarkdownToHtml = text => {
  var showdown = require('showdown')
  var converter = new showdown.Converter()
  const eventDescription = text.replace(/([ ]*\n)/g, `<br/>\n`)
  const finalDescription = eventDescription.replace(
    /(<br\/>\n<br\/>\n<br\/>)/g,
    `<br/><br/>`
  )
  return converter.makeHtml(finalDescription)
}
