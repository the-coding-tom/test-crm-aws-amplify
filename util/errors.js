export const displayError = function(e) {
  if (e.response) {
    processError(e.response.data.message)

    if (e.response.data.errors) {
      _.each(e.response.data.errors, (o, i) => {
        return _.each(o, (v, k) => processError(v))
      })
    }

    return
  }

  processError(e.message)
}

const processError = function(message) {
  console.log('------------------------------------')
  console.log(message)
  console.log('------------------------------------')
}
