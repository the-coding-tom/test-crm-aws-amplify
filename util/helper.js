const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

const parseRoomdate = (data, vm) => {
  const roomAvaila = data.split(',')
  const roomAvailability = []
  roomAvaila.forEach(data => {
    let weekday = []
    let timeof = []
    let lookup = {
      su: 0,
      mo: 1,
      tu: 2,
      we: 3,
      th: 4,
      fr: 5,
      sa: 6
    }
    const timeData = data.split(' ')
    if (timeData.length === 2) {
      weekday.push(timeData[0])
      timeof.push(timeData[1])
    }
    if (timeData.length === 3) {
      weekday.push(timeData[1])
      timeof.push(timeData[2])
    }
    const splitWeekday = weekday[0].split('-')
    const splitTimeOf = timeof[0].split('-')
    const AStruc = {
      from: '',
      to: '',
      weekdays: []
    }
    splitWeekday.forEach(week => {
      AStruc.weekdays.push(lookup[week])
    })
    AStruc.from = vm.$moment(splitTimeOf[0], 'HH').format('hh:mm')
    AStruc.to = vm.$moment(splitTimeOf[1], 'HH').format('hh:mm')
    roomAvailability.push(AStruc)
  })
  return roomAvailability
}

const undoParse = (data, vm) => {
  let str = ''
  const lookup = {
    0: 'su',
    1: 'mo',
    2: 'tu',
    3: 'we',
    4: 'th',
    5: 'fr',
    6: 'sa'
  }
  data.forEach(item => {
    let available = ''
    item.weekdays.forEach(day => {
      available += `${lookup[day]}-`
    })
    available = available.slice(0, -1)
    let weekdayfrom = item.from.charAt(0) + item.from.charAt(1)
    let weekdayto = item.to.charAt(0) + item.to.charAt(1)
    available += ` ${weekdayfrom}-${weekdayto}`
    str += `${available},`
  })
  str = str.slice(0, -1)
  return str
}

export default { randomColor, parseRoomdate, notify, parseRoomdate, undoParse }
