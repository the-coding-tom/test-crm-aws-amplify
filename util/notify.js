const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

export default notify
