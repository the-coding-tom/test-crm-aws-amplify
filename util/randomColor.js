const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

export default randomColor
