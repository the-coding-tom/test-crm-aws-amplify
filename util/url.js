export const getQueryParams = url => {
  const urlParam = require('url')

  const parsedUrl = urlParam.parse(url)

  return parsedUrl.search
}
