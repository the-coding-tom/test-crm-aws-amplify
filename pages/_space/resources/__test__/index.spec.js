import Card from '~/components/argon-core/Cards/Card.vue'
import Badge from '~/components/argon-core/Badge.vue'
import AddPage from '@/pages/_space/resources/add.vue'
import ViewPage from '@/pages/_space/resources/_id/index.vue'
import IndexPage from '@/pages/_space/resources/index.vue'
import BaseHeader from '~/components/argon-core/BaseHeader'
import BaseInput from '~/components/argon-core/Inputs/BaseInput.vue'
import BootstrapVue from 'bootstrap-vue'
import BasePagination from '~/components/argon-core/BasePagination'

import moment from 'moment'
import VueRouter from 'vue-router'
import cloneDeep from 'lodash/cloneDeep'
import flushPromises from 'flush-promises'
import { createWrapper } from '@/test.util.js'

import allResources from './resources/all-resources.json'

describe('Page: /:space/resources/ | Resources Index Page', () => {
  let NuxtStore, store, wrapper, $resource, router

  beforeAll(async () => {
    NuxtStore = await import(`${process.env.buildDir}/store.js`)
    const { createRouter } = await import(`${process.env.buildDir}/router.js`)

    createRouter()

    $resource = {
      getAllRooms: jest.fn().mockResolvedValue(cloneDeep(allResources)),
      deleteRoom: jest.fn().mockResolvedValue()
    }

    // mock confirm
    global.confirm = jest
      .fn()
      .mockReturnValueOnce(false)
      .mockReturnValue(true)
  })

  beforeEach(async () => {
    store = await NuxtStore.createStore()
    store.$resource = $resource
    store._vm.$bvToast = {
      toast: jest.fn().mockReturnValue()
    }

    router = new VueRouter({
      routes: [
        {
          path: '_space/resources/add',
          name: 'space-resources-add',
          component: AddPage
        },
        {
          path: '_space/resources/id/edit',
          name: 'space-resources-id-edit',
          component: ViewPage
        }
      ]
    })

    wrapper = createWrapper(
      IndexPage,
      store,
      router,
      {
        daylookup: {
          0: 'Sunday',
          1: 'Monday',
          2: 'Tuesday',
          3: 'Wednesday',
          4: 'Thursaday',
          5: 'Friday',
          6: 'Saturday'
        }
      },
      {
        stubs: {
          NuxtLink: true
        }
      },
      vueInstance => {
        vueInstance.use(BootstrapVue)
        vueInstance.component(BasePagination.name, BasePagination)
        vueInstance.component(BaseInput.name, BaseInput)
        vueInstance.component(BaseHeader.name, BaseHeader)
        vueInstance.component(Card.name, Card)
        vueInstance.component(Badge.name, Badge)
        vueInstance.prototype.$moment = moment
      }
    )

    await wrapper.vm.$options.asyncData({ store })
  })

  it('renders properly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders all resources', async () => {
    const wrapArr = wrapper.findAll('.card')

    expect(wrapArr.length).toBe(2)
  })

  it('reject removal a resource', async () => {
    wrapper.find('a.text-danger').trigger('click')
    await flushPromises()

    expect($resource.deleteRoom).not.toHaveBeenCalled()
  })

  it('removes a resource', async () => {
    wrapper.find('a.text-danger').trigger('click')
    await flushPromises()

    expect($resource.deleteRoom).toHaveBeenCalled()
  })
})
