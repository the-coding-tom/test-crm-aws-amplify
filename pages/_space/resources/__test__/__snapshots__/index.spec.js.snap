// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: /:space/resources/ | Resources Index Page renders properly 1`] = `
<div>
  <div class="header pb-6">
    <div class="container-fluid">
      <div class="header-body">
        <div class="d-flex justify-content-between py-4">
          <h1 class="page-title"><span>Resources</span> <b style="display: none;">â€¢</b> </h1>
          <nuxtlink-stub to="[object Object]" class="btn btn-primary">Add Resource</nuxtlink-stub>
        </div>
      </div>
    </div>
  </div>
  <div class="container-fluid mt--6">
    <div class="row equal">
      <div class="col-md-4">
        <div class="card sh-book-res">
          <!---->
          <div class="card-body">
            <div class="m-n25 img-wrap"><img src="https://d3mym26vamg9ol.cloudfront.net/resources/1572951811_5548b730-0d96-4897-92d8-90323785ef0a.jpeg" alt="" class="img-fluid"> <span class="badge badge-default">conference</span></div>
            <div class="mr-t-50">
              <div class="d-flex justify-content-between mr-b-20">
                <h3>Conference Room 1</h3>
                <h3 class="text-muted">20 USD/hr</h3>
              </div>
              <p class="text-muted">Some description of the room</p>
            </div>
            <div class="mr-t-30">
              <p><u>Resource Availability</u></p>
              <p>
                Monday - Saturday
                06:00 AM -
                05:00 AM</p>
            </div>
          </div>
          <!---->
          <div class="card-footer">
            <div>
              <div class="d-flex justify-content-between">
                <nuxtlink-stub to="[object Object]" class="edit-link"><i class="ti-pencil"></i> Edit Resource
                </nuxtlink-stub> <a class="text-danger"><i class="ti-trash"></i> Delete Resource</a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-md-4">
        <div class="card sh-book-res">
          <!---->
          <div class="card-body">
            <div class="m-n25 img-wrap"><img src="https://www.hourlyspaces.com/images/Friendspg_Banner.jpg" alt="" class="img-fluid"> <span class="badge badge-default">conference</span></div>
            <div class="mr-t-50">
              <div class="d-flex justify-content-between mr-b-20">
                <h3>Conference Rooms</h3>
                <h3 class="text-muted">10 USD/hr</h3>
              </div>
              <p class="text-muted">The conference room is a spacious and light filled room perfect for small to medium sized conferences. This has seats for 30 guests include wireless projection and is fully airconditioned.</p>
            </div>
            <div class="mr-t-30">
              <p><u>Resource Availability</u></p>
              <p>
                Monday - Saturday
                12:00 AM -
                11:00 PM</p>
            </div>
          </div>
          <!---->
          <div class="card-footer">
            <div>
              <div class="d-flex justify-content-between">
                <nuxtlink-stub to="[object Object]" class="edit-link"><i class="ti-pencil"></i> Edit Resource
                </nuxtlink-stub> <a class="text-danger"><i class="ti-trash"></i> Delete Resource</a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <ul class="pagination justify-content-center">
      <li class="page-item prev-page disabled"><a aria-label="Previous" class="page-link"><span aria-hidden="true"><i aria-hidden="true" class="fa fa-angle-left"></i></span></a></li>
      <li class="page-item active"><a class="page-link">1</a></li>
      <li class="page-item"><a class="page-link">2</a></li>
      <li class="page-item next-page"><a aria-label="Next" class="page-link"><span aria-hidden="true"><i aria-hidden="true" class="fa fa-angle-right"></i></span></a></li>
    </ul>
  </div>
</div>
`;
